src/wallet.rs

```diff
@@ -551,6 +557,92 @@ impl Wallet {
         Ok(false)
     }
 
+    // Creates a BMM request transaction. Does NOT broadcast. `height` is the block height this
+    // transaction MUST be included in. `amount` is the amount to spend in the BMM bid.
+    pub async fn create_bmm_request(
+        &self,
+        sidechain_number: SidechainNumber,
+        sidechain_block_hash: bdk::bitcoin::BlockHash,
+        prev_mainchain_block_hash: bdk::bitcoin::BlockHash,
+        amount: bdk::bitcoin::Amount,
+        locktime: bdk::bitcoin::absolute::LockTime,
+    ) -> Result<bdk::bitcoin::Transaction> {
+        let mut psbt = self.build_bmm_tx(
+            sidechain_number,
+            sidechain_block_hash,
+            prev_mainchain_block_hash,
+            amount,
+            locktime,
+        )?;
+
+        let psbt_finalized: Result<bool, _> = self
+            .bitcoin_wallet
+            .lock()
+            .sign(&mut psbt, SignOptions::default());
+        match psbt_finalized {
+            Ok(true) => {
+                tracing::info!("BMM request psbt signed successfully");
+            }
+
```

Comment:
Why is there so much whitespace?

Reply:
I guess this boils down to personal preference, but I find blocks of code much easier to parse when there's a little breathing room between them. Ideally the code formatter would handle absolutely everything, and there'd be no ambiguity in formatting. But whitespace is one of those things that formatters unfortunately aren't handling 100% strictly...